From 03397a537009fbd2fb6831344810638b994baf09 Mon Sep 17 00:00:00 2001
From: Eric Richter <erichte@linux.ibm.com>
Date: Wed, 14 Aug 2019 12:27:53 -0500
Subject: [PATCH] discover: force kexec_file_load

---
 discover/boot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/discover/boot.c b/discover/boot.c
index 91fc46d..ac02f37 100644
--- a/discover/boot.c
+++ b/discover/boot.c
@@ -106,7 +106,7 @@ static int kexec_load(struct boot_task *boot_task)
 
 	p = argv;
 	*p++ = pb_system_apps.kexec;	/* 1 */
-	*p++ = "-l";			/* 2 */
+	*p++ = "-s";			/* 2 */
 
 	if (pb_log_get_debug()) {
 		*p++ = "--debug";	/* 3 */
-- 
2.20.1

